app/src/ai/components/ai-message-list.vue

```diff
@@ -0,0 +1,391 @@
+<script setup lang="ts">
+import type { ComponentPublicInstance } from 'vue';
+import { ref, watch, nextTick, onMounted, onBeforeUnmount } from 'vue';
+import { useElementBounding } from '@vueuse/core';
+import AiMessage from './ai-message.vue';
+
+import type { UIMessage } from 'ai';
+
+export type AiMessage = UIMessage;
+
+export type AiStatus = 'ready' | 'streaming' | 'submitted' | 'error';
+
+interface Props {
+	/** Array of messages to display */
+	messages?: UIMessage[];
+	/** Current status of the AI interaction */
+	status?: AiStatus;
+	/** Auto-scroll to bottom when streaming */
+	shouldAutoScroll?: boolean;
```

Comment:
Isn't this always true in our case as we're only using this component in one place? Feels like we're making this a bit too configurable where we don't really use a lot of these options